gm4_speed_paths/overlay_26/data/gm4_speed_paths/functions/apply_effects/path_check.mcfunction

```diff
@@ -0,0 +1,8 @@
+# checks if the player is (still) on a speed path, otherwise break
+# @s = player
+# at @s
+# run from main
+
+execute if entity @s[tag=!gm4_on_path] unless function gm4_speed_paths:apply_effects/add_speed run return 0
```

Comment:
Having the `if function` make changes to the state of the player is somewhat weird (we should probably restrict ourselves to using `if function` with pure functions only, or close-to).

Instead of the `unless function` use 
```suggestion
execute if entity @s[tag=!gm4_on_path] return run function gm4_speed_paths:apply_effects/add_speed
```
and adjust the returns within the add speed function accordingly.